feat: folks xchain

palace22 opened the pull request:
Added Folks Finance xChain yields:
- base APY from lending
- rewards APY from reward V1 contract

llamatester commented:
The folks-finance-xchain adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       157 passed, 157 total
    Snapshots:   0 total
    Time:        0.27 s
    Ran all test suites.
    
    Nb of pools: 17
     
    
    Sample pools:
    ┌─────────┬───────────────────────────────────────────────────────┬─────────────┬────────────────────────┬───────────┬────────────────────┬──────────────────────────────────────────────────┬───────────────────────────────────────────────────────┬──────────────────────┬─────────┬───────────────┬────────────────────┬────────────────────┐
    │ (index) │ pool                                                  │ chain       │ project                │ symbol    │ tvlUsd             │ underlyingTokens                                 │ rewardTokens                                          │ apyReward            │ apyBase │ apyBaseBorrow │ totalSupplyUsd     │ totalBorrowUsd     │
    ├─────────┼───────────────────────────────────────────────────────┼─────────────┼────────────────────────┼───────────┼────────────────────┼──────────────────────────────────────────────────┼───────────────────────────────────────────────────────┼──────────────────────┼─────────┼───────────────┼────────────────────┼────────────────────┤
    │ 0       │ '0x307bcec89624660ed06c97033edb7ef49ab0eb2d-avax'     │ 'Avalanche' │ 'folks-finance-xchain' │ 'SolvBTC' │ 2828563.19269145   │ [ '0xbc78D84Ba0c46dFe32cf2895a19939c86b81a777' ] │ [ 'avax:0x0000000000000000000000000000000000000000' ] │ 0.029540810311936082 │ 0       │ 0.0005        │ 2839305.5504289325 │ 10742.357737483308 │
    │ 1       │ '0x88f15e36308ed060d8543da8e2a5da0810efded2-avax'     │ 'Avalanche' │ 'folks-finance-xchain' │ 'USDC'    │ 1205680.0864594255 │ [ '0xb97ef9ef8734c71904d8002f8b6bc66dd9c48a6e' ] │ [ 'avax:0x0000000000000000000000000000000000000000' ] │ 0.07450904328599571  │ 0.04867 │ 0.06745       │ 6303997.220421297  │ 5099407.598173523  │
    │ 2       │ '0x795ccf6f7601edb41e4b3123c778c56f0f19389a-avax'     │ 'Avalanche' │ 'folks-finance-xchain' │ 'WETH.e'  │ 1036321.0098703629 │ [ '0x49d5c2bdffac6ce2bfdb6640f4f80f226bc10bab' ] │ [ 'avax:0x0000000000000000000000000000000000000000' ] │ 0.047046299690272374 │ 0.00147 │ 0.01802       │ 1140919.2577280523 │ 104598.24785768929 │
    │ 3       │ '0x0259617be41ada4d97ded60daf848caa6db3f228-avax'     │ 'Avalanche' │ 'folks-finance-xchain' │ 'AVAX'    │ 1031554.918568687  │ [ '0x0000000000000000000000000000000000000000' ] │ [ 'avax:0x0000000000000000000000000000000000000000' ] │ 0.055180251345962765 │ 0.00206 │ 0.02057       │ 1161936.7932119982 │ 130382.1274433111  │
    │ 4       │ '0x1c51aa1516e1156d98075f2f64e259906051aba9-avax'     │ 'Avalanche' │ 'folks-finance-xchain' │ 'BTC.b'   │ 313550.622033      │ [ '0x152b9d0fdc40c096757f570a51e494bd4b943e50' ] │ [ 'avax:0x0000000000000000000000000000000000000000' ] │ 0.024150258728091088 │ 0.00129 │ 0.01321       │ 352055.01804       │ 38504.396007       │
    │ 5       │ '0x51958ed7b96f57142ce63bb223bbd9ce23da7125-base'     │ 'Base'      │ 'folks-finance-xchain' │ 'ETH'     │ 87537.77609015112  │ [ '0x0000000000000000000000000000000000000000' ] │ []                                                    │ 0                    │ 0.00171 │ 0.01889       │ 97271.39233250126  │ 9891.191603204941  │
    │ 6       │ '0xb6df8914c084242a19a4c7fb15368be244da3c75-ethereum' │ 'Ethereum'  │ 'folks-finance-xchain' │ 'ETH'     │ 74495.57287424554  │ [ '0x0000000000000000000000000000000000000000' ] │ []                                                    │ 0                    │ 0       │ 0.01005       │ 74496.11494624555  │ 0.542072           │
    │ 7       │ '0x9936812835476504d6cf495f4f0c718ec19b3aff-ethereum' │ 'Ethereum'  │ 'folks-finance-xchain' │ 'WBTC'    │ 27932.789052       │ [ '0x2260fac5e5542a773aa44fbcfedf7c193bc2c599' ] │ []                                                    │ 0                    │ 0       │ 0             │ 27932.789052       │ 0                  │
    │ 8       │ '0x5e5a2007a8d613c4c98f425097166095c875e6ee-avax'     │ 'Avalanche' │ 'folks-finance-xchain' │ 'JOE'     │ 20392.364868976285 │ [ '0x6e84a6216ea6dacc71ee8e6b0a5b7322eebc0fdd' ] │ [ 'avax:0x0000000000000000000000000000000000000000' ] │ 0.08984342010939618  │ 0.00285 │ 0.03113       │ 22741.44383699621  │ 2349.0789680199223 │
    │ 9       │ '0x1177a3c2cccdb9c50d52fc2d30a13b2c3c40bcf4-arbitrum' │ 'Arbitrum'  │ 'folks-finance-xchain' │ 'ARB'     │ 18437.754759256106 │ [ '0x912ce59144191c1204e64559fe8253a0e49e6548' ] │ []                                                    │ 0                    │ 0.00037 │ 0.00707       │ 19588.288382090133 │ 1150.5336228340286 │
    └─────────┴───────────────────────────────────────────────────────┴─────────────┴────────────────────────┴───────────┴────────────────────┴──────────────────────────────────────────────────┴───────────────────────────────────────────────────────┴──────────────────────┴─────────┴───────────────┴────────────────────┴────────────────────┘
    This adapter contains some pools with <10k TVL, these pools won't be shown in DefiLlama

slasher125 commented:
gm, ty for adding
- make sure to scale apy values to %
- rewardTokens pls remove the prefix eg avax
- if you add ltv we can display your adapter on our borrow aggregator too

llamatester commented:
The folks-finance-xchain adapter exports pools: 
        
 
 Test Suites: 1 passed, 1 total
    Tests:       174 passed, 174 total
    Snapshots:   0 total
    Time:        0.294 s
    Ran all test suites.
    
    Nb of pools: 17
     
    
    Sample pools:
    ┌─────────┬───────────────────────────────────────────────────────┬─────────────┬────────────────────────┬───────────┬────────────────────┬──────────────────────────────────────────────────┬──────────────────────────────────────────────────┬───────────────────┬─────────────────────┬────────────────────┬────────────────────┬────────────────────┬───────┐
    │ (index) │ pool                                                  │ chain       │ project                │ symbol    │ tvlUsd             │ underlyingTokens                                 │ rewardTokens                                     │ apyReward         │ apyBase             │ apyBaseBorrow      │ totalSupplyUsd     │ totalBorrowUsd     │ ltv   │
    ├─────────┼───────────────────────────────────────────────────────┼─────────────┼────────────────────────┼───────────┼────────────────────┼──────────────────────────────────────────────────┼──────────────────────────────────────────────────┼───────────────────┼─────────────────────┼────────────────────┼────────────────────┼────────────────────┼───────┤
    │ 0       │ '0x307bcec89624660ed06c97033edb7ef49ab0eb2d-avax'     │ 'Avalanche' │ 'folks-finance-xchain' │ 'SolvBTC' │ 2798796.8924765214 │ [ '0xbc78D84Ba0c46dFe32cf2895a19939c86b81a777' ] │ [ '0x0000000000000000000000000000000000000000' ] │ 3.048106755821384 │ 0                   │ 0.03               │ 2805057.129565182  │ 6260.237088660565  │ 0.65  │
    │ 1       │ '0x88f15e36308ed060d8543da8e2a5da0810efded2-avax'     │ 'Avalanche' │ 'folks-finance-xchain' │ 'USDC'    │ 1506308.4455485241 │ [ '0xb97ef9ef8734c71904d8002f8b6bc66dd9c48a6e' ] │ [ '0x0000000000000000000000000000000000000000' ] │ 7.458010082800717 │ 4.417999999999999   │ 6.4750000000000005 │ 6420064.373856101  │ 4914846.353262321  │ 0.8   │
    │ 2       │ '0x0259617be41ada4d97ded60daf848caa6db3f228-avax'     │ 'Avalanche' │ 'folks-finance-xchain' │ 'AVAX'    │ 1074932.3759312304 │ [ '0x0000000000000000000000000000000000000000' ] │ [ '0x0000000000000000000000000000000000000000' ] │ 5.469536688337012 │ 0.174               │ 1.9460000000000002 │ 1194958.9531366732 │ 120026.83490544268 │ 0.725 │
    │ 3       │ '0x795ccf6f7601edb41e4b3123c778c56f0f19389a-avax'     │ 'Avalanche' │ 'folks-finance-xchain' │ 'WETH.e'  │ 1010178.0070499589 │ [ '0x49d5c2bdffac6ce2bfdb6640f4f80f226bc10bab' ] │ [ '0x0000000000000000000000000000000000000000' ] │ 4.919815053040559 │ 0.148               │ 1.802              │ 1112164.308044372  │ 101986.30099441303 │ 0.8   │
    │ 4       │ '0x1c51aa1516e1156d98075f2f64e259906051aba9-avax'     │ 'Avalanche' │ 'folks-finance-xchain' │ 'BTC.b'   │ 312084.31984416    │ [ '0x152b9d0fdc40c096757f570a51e494bd4b943e50' ] │ [ '0x0000000000000000000000000000000000000000' ] │ 2.473275628249357 │ 0.129               │ 1.322              │ 350426.6748792     │ 38342.35503504     │ 0.725 │
    │ 5       │ '0x51958ed7b96f57142ce63bb223bbd9ce23da7125-base'     │ 'Base'      │ 'folks-finance-xchain' │ 'ETH'     │ 83563.23470120865  │ [ '0x0000000000000000000000000000000000000000' ] │ []                                               │ 0                 │ 0.17600000000000002 │ 1.9060000000000001 │ 93055.4967732636   │ 9645.923571543417  │ 0.8   │
    │ 6       │ '0xb6df8914c084242a19a4c7fb15368be244da3c75-ethereum' │ 'Ethereum'  │ 'folks-finance-xchain' │ 'ETH'     │ 72651.54011773145  │ [ '0x0000000000000000000000000000000000000000' ] │ []                                               │ 0                 │ 0                   │ 1.005              │ 72652.06872573144  │ 0.528608           │ 0.8   │
    │ 7       │ '0x9936812835476504d6cf495f4f0c718ec19b3aff-ethereum' │ 'Ethereum'  │ 'folks-finance-xchain' │ 'WBTC'    │ 27790.84021602     │ [ '0x2260fac5e5542a773aa44fbcfedf7c193bc2c599' ] │ []                                               │ 0                 │ 0                   │ 0                  │ 27790.84021602     │ 0                  │ 0.75  │
    │ 8       │ '0x5e5a2007a8d613c4c98f425097166095c875e6ee-avax'     │ 'Avalanche' │ 'folks-finance-xchain' │ 'JOE'     │ 24095.31066902902  │ [ '0x6e84a6216ea6dacc71ee8e6b0a5b7322eebc0fdd' ] │ [ '0x0000000000000000000000000000000000000000' ] │ 7.862304278526009 │ 0.22899999999999998 │ 2.848              │ 26490.60102787564  │ 2395.290358846621  │ 0.45  │
    │ 9       │ '0x1177a3c2cccdb9c50d52fc2d30a13b2c3c40bcf4-arbitrum' │ 'Arbitrum'  │ 'folks-finance-xchain' │ 'ARB'     │ 18979.73638947532  │ [ '0x912ce59144191c1204e64559fe8253a0e49e6548' ] │ []                                               │ 0                 │ 0.037               │ 0.707              │ 20164.09018709026  │ 1184.3537976149391 │ 0.65  │
    └─────────┴───────────────────────────────────────────────────────┴─────────────┴────────────────────────┴───────────┴────────────────────┴──────────────────────────────────────────────────┴──────────────────────────────────────────────────┴───────────────────┴─────────────────────┴────────────────────┴────────────────────┴────────────────────┴───────┘
    This adapter contains some pools with <10k TVL, these pools won't be shown in DefiLlama

palace22 commented:
> gm, ty for adding
> 
> * make sure to scale apy values to %
> * rewardTokens pls remove the prefix eg avax
> * if you add ltv we can display your adapter on our borrow aggregator too

Gm, thanks, I scaled apy to % and added ltv. 
Regarding the reward token I removed the prefix but we will add reward tokens from different chains in the near future, how will I be able to add them?

slasher125 commented:
> Regarding the reward token I removed the prefix but we will add reward tokens from different chains in the near future, how will I be able to add them?

just use the address from that chain, the rest is handled by us (the reward token addresses are mostly used for displaying icons on our ui)
